src/compiler/checker.ts

```diff
@@ -27861,6 +27889,7 @@ namespace ts {
         }
 
         function getSuggestedSymbolForNonexistentProperty(name: Identifier | PrivateIdentifier | string, containingType: Type): Symbol | undefined {
+            const originalName = name;

```

Comment:
this would be more elegant to have 

1. the parameter change from `name -> propertyName`
2. the `if` after `let props ...` become

```
let name;
if (typeof propertyName === 'string') {
  name = propertyName
} else {
  name = idText(propertyName)
  const parent = name.parent;
  // ...
}
```